### PR TITLE
[Merged by Bors] - feat: port Topology.Alexandroff

### PR DESCRIPTION
**Pull request by urkud**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Alexandroff.lean [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+    ∀ z : Alexandroff X, C z
+  | ∞ => h₁
+  | (x : X) => h₂ x
+#align alexandroff.rec Alexandroff.rec
+
+theorem isCompl_range_coe_infty : IsCompl (range ((↑) : X → Alexandroff X)) {∞} :=
+  isCompl_range_some_none X
+#align alexandroff.is_compl_range_coe_infty Alexandroff.isCompl_range_coe_infty
+
+-- porting note: moved @[simp] to a new lwmma
```
Comment (RemyDegenne) [inline Mathlib/Topology/Alexandroff.lean:120]: Could you write to which lemma you moved it?
```suggestion
-- porting note: moved @[simp] to a new lemma
```

File: Mathlib/Topology/Alexandroff.lean [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+theorem ne_infty_iff_exists {x : Alexandroff X} : x ≠ ∞ ↔ ∃ y : X, (y : Alexandroff X) = x := by
+  induction x using Alexandroff.rec <;> simp
+#align alexandroff.ne_infty_iff_exists Alexandroff.ne_infty_iff_exists
+
+instance canLift : CanLift (Alexandroff X) X (↑) fun x => x ≠ ∞ :=
+  WithTop.canLift
+#align alexandroff.can_lift Alexandroff.canLift
+
+theorem not_mem_range_coe_iff {x : Alexandroff X} : x ∉ range ((↑) : X → Alexandroff X) ↔ x = ∞ :=
+  by rw [← mem_compl_iff, compl_range_coe, mem_singleton_iff]
```
Comment (RemyDegenne) [inline Mathlib/Topology/Alexandroff.lean:156]: Is that `by` at the beginning of a line something we accept in the new convention? (I don't know)
If not:
```suggestion
theorem not_mem_range_coe_iff {x : Alexandroff X} :
    x ∉ range ((↑) : X → Alexandroff X) ↔ x = ∞ := by
  rw [← mem_compl_iff, compl_range_coe, mem_singleton_iff]
```

**Review by RemyDegenne — APPROVED** [review]
Thanks!
bors d+

File: Mathlib/Topology/Alexandroff.lean [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+    TopologicalSpace.Opens (Alexandroff X) :=
+  ⟨((↑) '' s)ᶜ, isOpen_compl_image_coe.2 ⟨h₁, h₂⟩⟩
+#align alexandroff.opens_of_compl Alexandroff.opensOfCompl
+
+theorem infty_mem_opensOfCompl {s : Set X} (h₁ : IsClosed s) (h₂ : IsCompact s) :
+    ∞ ∈ opensOfCompl s h₁ h₂ :=
+  mem_compl infty_not_mem_image_coe
+#align alexandroff.infty_mem_opens_of_compl Alexandroff.infty_mem_opensOfCompl
+
+-- porting note: todo: add @[continuity]
```
Comment (mcdoll) [inline Mathlib/Topology/Alexandroff.lean:266]:
```suggestion
@[continuity]
```
and merge master

File: Mathlib/Topology/Alexandroff.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Topology/Alexandroff.lean]: I used `some` to save a line.

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**